### ISSUE TITLE
[Accessibility] Tree implementation should conform to WAI-ARIA 1.2 design pattern

### ISSUE THREAD

**Issue by majornista** [adobe/coral-spectrum#10]
### Expected Behavior
Per: https://jira.corp.adobe.com/browse/CQ-4272964

Coral-Spectrum Tree implementation should conform with WAI-ARIA Authoring Practice design pattern: https://www.w3.org/TR/wai-aria-practices-1.2/#TreeView

- ArrowRight should expand focused "drilldown" node.
- ArrowLeft should collapse focused "drilldown" node.
- Use aria-expanded and aria-owns to properly express the hierarchical relationship of a parent item to its subTreeContainer
- Improve labelling of nodes to better communicate selected or unselected state.
- Improve labelling of subTreeContainer.
- On mobile devices provide mechanism to expand or collapse and to toggle selection using a mobile screen reader.

### Actual Behavior
- ArrowRight navigates to next item
- ArrowLeft navigated to previous item
- Expanded state is not communicated to assistive technology, and the tree hierarchy is incorrect, because each subTreeContainer follows its parent item in the DOM with now aria-owns property to express the hierarchical behavior.
- On mobile there is no way to expand/collapse independent of select/deselect using a mobile screen reader. The button to expand or collapse should be independent from the mechanism to select or deselect.

#### Browser name/version/os (ie Chrome Version 62.0.3202.94 (Official Build) (64-bit) MacOS)
All

#### Coral Spectrum version
v4.0.0
